File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int quater = 25;
int dime = 10;
int nickel = 5;
int penny = 1;

int sum;

int main(void)
{

    float dollars;
    int coins = 0;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);
    int cents = round(dollars * 100);

    do
    {
        if (cents >= quater)
        {
            cents = cents - quater;
            coins++;
        }
    }
    while (cents >= quater);

    do
    {
        if (cents >= dime)
        {
            cents = cents - dime;
            coins++;
        }
    }
    while (cents >= dime);

    do
    {
        if (cents >= nickel)
        {
            cents = cents - nickel;
            coins++;
        }
    }
    while (cents >= nickel);

    do
    {
        if (cents >= penny)
        {
            cents = cents - penny;
            coins++;
        }
    }
    while (cents >= penny);

    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float

quater = 25
dime = 10
nickel = 5
penny = 1


def main():
    # get correct value from input
    while True:
        dollars = get_float("Change owed: ")
        if dollars > 0:
            break
    # convert value into cents
    cents = int(dollars * 100)
    # init counting function with our value
    countCouns(cents)


# fuction: count coints in each type
def countCouns(cents):
    # set initial total coins to zero
    total = 0

    # check each type in array
    for coin in [quater, dime, nickel, penny]:
        # do if is in max value
        while cents >= coin:
            cents = cents - coin
            total += 1
    # print the result
    print(total)


main()
